src/enzo/CallInSitulibyt.C

```diff
@@ -149,10 +164,11 @@ int CallInSitulibyt(LevelHierarchyEntry *LevelArray[], TopGridData *MetaData,
     params->length_unit = LengthUnits;
     params->mass_unit = DensityUnits * LengthUnits * LengthUnits * LengthUnits; /* this right? */
     params->time_unit = TimeUnits;
-    params->magnetic_unit = 0.0; /* Not right */
-    params->periodicity[0] = 1; /* also not right */
-    params->periodicity[1] = 1;
-    params->periodicity[2] = 1;
+    params->velocity_unit = VelocityUnits;
+    params->magnetic_unit = sqrt(4.0 * 3.141592653589793238462643383279502884L * DensityUnits) * VelocityUnits;
+    params->periodicity[0] = MetaData->LeftFaceBoundaryCondition[0];
+    params->periodicity[1] = MetaData->LeftFaceBoundaryCondition[1];
+    params->periodicity[2] = MetaData->LeftFaceBoundaryCondition[2];
```

Comment:
@matthewturk Please double-check this to see if I'm doing it correctly. Thanks!

Reply:
The values are here:

```
const boundary_type
  reflecting        = 0,
  outflow           = 1,
  inflow            = 2,
  periodic          = 3,
  shearing          = 4,
  hydrostatic       = 5,
  BoundaryUndefined = 6;
```

so what you should do is compare to `periodic` to see if it's that.  We only allow this to be True or False I think.

Reply:
i.e.,

```
params->periodicity[0] = (MetaData->LeftFaceBoundaryCondition[0] == periodic);
```